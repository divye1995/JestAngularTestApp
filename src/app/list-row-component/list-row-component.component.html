<div [ngClass]="getClass()">
    <span>{{firstName}} {{lastName}}</span> <button data-testid="row-click" (click)="rowClick.emit()">o</button>
</div>