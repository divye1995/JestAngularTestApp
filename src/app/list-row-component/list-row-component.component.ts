import { Component, EventEmitter, Input, OnInit, Output, } from '@angular/core';

@Component({
  selector: 'app-list-row-component',
  templateUrl: './list-row-component.component.html',
  styleUrls: ['./list-row-component.component.scss'],

})
export class ListRowComponentComponent implements OnInit {

  @Input() firstName:string;
  @Input() lastName:string;
  @Input() gender:string;
  @Output() rowClick = new EventEmitter();

  getClass(){
    return {
      'blue':this.gender==='male',
      'green':this.gender==='female'
    }
  }
  constructor() { 
  }
  ngOnInit() {
  }
}
