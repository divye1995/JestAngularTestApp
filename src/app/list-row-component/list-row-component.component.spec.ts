import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { getByTestId } from '@testing-library/dom'
 
import { ListRowComponentComponent } from './list-row-component.component';
import { DebugElement } from '@angular/core';
import { simulateClick } from '../testing/event.helper';


describe('ListRowComponentComponent', () => {
  let component: ListRowComponentComponent;
  let fixture: ComponentFixture<ListRowComponentComponent>;

  
  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ ListRowComponentComponent ]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(ListRowComponentComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should render the component with blue color class',()=>{
    component.firstName = 'James'
    component.lastName = 'Bond'
    component.gender = 'male'
    fixture.detectChanges()

    expect(fixture).toMatchSnapshot();
  })
  it('should render the component with green color class',()=>{
    component.firstName = 'James'
    component.lastName = 'Bond'
    component.gender = 'female'
    fixture.detectChanges()

    expect(fixture).toMatchSnapshot();
  })

  it('should emit events onClick',done=>{
    let buttonClicked = false
    component.rowClick.subscribe(()=>{
      buttonClicked =true;
      expect(buttonClicked).toBeTruthy()
      done();
    })
    var btn = getByTestId(fixture.nativeElement,'row-click');
    simulateClick(btn);
  })
});
