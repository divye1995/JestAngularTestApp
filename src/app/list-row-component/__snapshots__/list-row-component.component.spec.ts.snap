// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListRowComponentComponent should render the component with blue color class 1`] = `
<app-list-row-component
  firstName={[Function String]}
  gender={[Function String]}
  lastName={[Function String]}
  rowClick={[Function EventEmitter]}
>
  <div
    class="blue"
    ng-reflect-ng-class="[object Object]"
  >
    <span>
      James Bond
    </span>
    <button
      data-testid="row-click"
    >
      o
    </button>
  </div>
</app-list-row-component>
`;

exports[`ListRowComponentComponent should render the component with green color class 1`] = `
<app-list-row-component
  firstName={[Function String]}
  gender={[Function String]}
  lastName={[Function String]}
  rowClick={[Function EventEmitter]}
>
  <div
    class="green"
    ng-reflect-ng-class="[object Object]"
  >
    <span>
      James Bond
    </span>
    <button
      data-testid="row-click"
    >
      o
    </button>
  </div>
</app-list-row-component>
`;
